src/components/CartCard/CartCard.jsx

```diff
@@ -0,0 +1,88 @@
+import "./cart-card.css";
+import { Link } from "react-router-dom";
+import { useCartContext } from "../../context/cart-context";
+import { useWishlistContext } from "../../context/wishlist-context";
+
+const CartCard = ({ products }) => {
+  const { cartDispatch } = useCartContext();
+  const { wishlistState, wishlistDispatch } = useWishlistContext();
+
+  return (
+    <div className="saiyan-horizontal-card">
+      <div className="card__img__container">
+        <img className="card__img" src={products.img} alt="Mens Tshirt" />
+      </div>
+
+      <div className="saiyan-horizontal-card__body">
+        <h3>{products.name}</h3>
+        <h1 className="card__price spacing">
+          Rs. {products.discountedPrice}
+          <span className="card__price__stricked">Rs. {products.price}</span>
+        </h1>
+        <h2 className="card__price__discount">
+          {products.discountInPercentage}% off
+        </h2>
+
+        <div className="card__quantity">
+          <span>Quantity:</span>
+          <span className="spacing">
+            <button
+              onClick={() =>
```

Comment:
You can add functionality that user should not decrement if there is only one product in that case make button disabled.

Reply:
implemented that functionality in cartReducer function